### PR TITLE
Add identity default UI

### PR DESCRIPTION
**Pull request by Yvonne-Wong**
We have added Identity to the front end endpoints and the back  end. The Identity database is also set up :)

### CODE REVIEW

File: ClientApp/src/components/api-authorization/AuthorizeRoute.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    }
+                }} />
+        }
+    }
+
+    async populateAuthenticationState() {
+        const authenticated = await authService.isAuthenticated();
+        this.setState({ ready: true, authenticated });
+    }
+
```
Comment (littlemauli) [inline ClientApp/src/components/api-authorization/AuthorizeRoute.js:51]: maybe an extra blank space to get rid off

File: ClientApp/src/components/api-authorization/AuthorizeService.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+                    return this.error("The user closed the window.");
+                } else if (!this._popUpDisabled) {
+                    console.log("Popup authentication error: ", popUpError);
+                }
+
+                // PopUps might be blocked by the user, fallback to redirect
+                try {
+                    await this.userManager.signinRedirect(this.createArguments(state));
+                    return this.redirect();
+                } catch (redirectError) {
```
Comment (littlemauli) [inline ClientApp/src/components/api-authorization/AuthorizeService.js]: the try-catch brackets don't align , but I am not sure that they should.

File: Models/Database/AdminUser.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using whale_spotting.Models.ApiModels;
+using Microsoft.AspNetCore.Identity;
+
+namespace whale_spotting.Models.Database
+{
+    public class AdminUser : IdentityUser
+    {
+    }
```
Comment (littlemauli) [inline Models/Database/AdminUser.cs:11]: Should this be empty?

File: Repositories/AdminRepo.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    public class AdminRepo : IAdminRepo
+    {
+        private readonly WhaleSpottingContext _context;
+
+        public AdminRepo(WhaleSpottingContext context)
+        {
+            _context = context;
+        }
+
+
```
Comment (littlemauli) [inline Repositories/AdminRepo.cs]: empty spaces

**Review by littlemauli — COMMENTED** [review]
not much to comment , looks really ggod

File: Startup.cs [hunk 5]
```diff
@@ -46,11 +65,11 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
+                app.UseMigrationsEndPoint(); //
```
Comment (ChristianMcc) [inline Startup.cs]: couple of slashes here

**Review by ChristianMcc — APPROVED** [review]
Looking great! Couple of formatting things to change then you can merge

File: Models/Database/AdminUser.cs (hunk 3 above)
Comment (chloejonessoftwire) [reply Models/Database/AdminUser.cs:11]: yes

Comment (Yvonne-Wong) [reply Models/Database/AdminUser.cs:11]: Yes, this is intentionally empty :)

**Pull request merged**